app/app.js

```diff
@@ -0,0 +1,56 @@
+import {App, IonicApp, Platform} from 'ionic/ionic';
+
+import {HelloIonicPage} from './pages/hello-ionic/hello-ionic';
+import {ListPage} from './pages/list/list';
+
+@App({
+  templateUrl: 'build/app.html'
+})
+class MyApp {
+  constructor(app: IonicApp, platform: Platform) {
+
+    // set up our app
+    this.app = app;
+    this.platform = platform;
+    this.initializeApp();
+
+    // set our app's pages
+    this.pages = [
+      { title: 'Hello Ionic', component: HelloIonicPage },
+      { title: 'My First List', component: ListPage }
+    ];
+
+    // make HelloIonicPage the root (or first) page
+    this.rootPage = HelloIonicPage;
+  }
+
+  initializeApp() {
+    this.platform.ready().then(() => {
+      console.log('Platform ready');
+
+      // The platform is now ready. Note: if this callback fails to fire, follow
+      // the Troubleshooting guide for a number of possible solutions:
+      //
+      // Okay, so the platform is ready and our plugins are available.
+      // Here you can do any higher level native things you might need.
+      //
+      // First, let's hide the keyboard accessory bar (only works natively) since
+      // that's a better default:
+      //
+      //
+      // For example, we might change the StatusBar color. This one below is
+      // good for light backgrounds and dark text;
+      if (window.StatusBar) {
+        window.StatusBar.styleDefault();
+      }
+    });
+  }
+
+  openPage(page) {
+    // close the menu when clicking a link from the menu
+    this.app.getComponent('leftMenu').close();
+    // navigate to the new page if it is not the current page
+    let nav = this.app.getComponent('nav');
+    nav.setRoot(page.component);
+  }
```

Comment:
Outside of the constructor we define a function called `openPage` that takes in a `page` as a parameter, and will set that as the current page (we will make use of this from outside of this component soon). Here we see a new function called `getComponent`, this is provided by the `IonicApp` service – it takes in an `id` and will return the component in the app with that id. In this case we are using it to grab the menu component and close it whenever a page is selected, and also to grab the nav component so that we can change the rootPage (by calling `setRoot`).